src/Form/DataTransformer/ModelsToArrayTransformer.php

```diff
@@ -127,7 +126,7 @@ public function transform($collection)
 
         $array = [];
         foreach ($collection as $key => $entity) {
-            $id = implode(AdapterInterface::ID_SEPARATOR, $this->getIdentifierValues($entity));
+            $id = $this->getNormalizedIdentifier($entity);
```

Comment:
Can this be replaced by `$this->modelManager->getNormalizedIdentifier($entity);` ?